Landing page

DinDin985 opened the pull request:
WIP

BBruington commented on apps/chameleon/components/ui/button.tsx:
```diff
@@ -17,6 +17,8 @@ const buttonVariants = cva(
         secondary:
           'bg-secondary text-secondary-foreground hover:bg-secondary/80',
         ghost: 'hover:bg-accent hover:text-accent-foreground',
+        aware: 'bg-white text-[#191919] hover:bg-[#e5e5e5]',
```
What is an aware button?

Balance8 replied:
the name aware is not a clear variant name

Balance8 commented on apps/chameleon/app/(auth)/register/page.tsx:
```diff
@@ -1,44 +1,138 @@
+import TypeWritterEffect from '@/components/signuppage/typing'
 import { UserAuthForm } from '@/components/user-auth-form'
 
+import chameleon from '../../../public/icons/chameleon-logo-transparent-bg.png'
```
These imports need to be updated to use @/

Balance8 commented on apps/chameleon/app/(auth)/register/page.tsx:
```diff
@@ -1,44 +1,138 @@
         href="/login"
         className={cn(
-          buttonVariants({ variant: 'ghost' }),
+          buttonVariants({ variant: 'aware' }),
```
lets not make a variant for this until its used more then 3 times

DinDin985 commented:
Note: Need to figure out the proper way to import images.

Balance8 commented on apps/chameleon/app/(auth)/register/page.tsx:
```diff
@@ -1,44 +1,138 @@
-          <div className="flex flex-col space-y-2 text-center">
-            <Icons.logo className="mx-auto h-6 w-6" />
-            <h1 className="text-2xl font-semibold tracking-tight">
+        <div className="mx-auto flex w-full flex-col justify-center space-y-6 sm:w-[350px] text-[#191919]">
```
we need to avoid text color arb values. these need to be brought into the design system color theme and then re written as text-insert-theme-color

DinDin985 replied:
I added the arb values to the theming, thoughts on changing these values to the default black and white in order to help with eye-strain?
![AddedColorsToThemeJPG](https://github.com/Ionia-Devs/chameleon/assets/100187027/f2bee35f-ffce-461e-a570-e1e5d73c2449)

Balance8 replied:
Lets remove these and use tailwind default color theme

Balance8 commented on apps/chameleon/app/global.css:
This does not need to be copied over, the global styles already exist at apps/chameleon/styles/globals.css

Balance8 commented on apps/chameleon/components/ui/button.tsx:
```diff
@@ -17,6 +17,8 @@ const buttonVariants = cva(
           'bg-secondary text-secondary-foreground hover:bg-secondary/80',
         ghost: 'hover:bg-accent hover:text-accent-foreground',
+        aware: 'bg-white text-[#191919] hover:bg-[#e5e5e5]',
+        signup: "",
```
remove this

Balance8 commented on apps/chameleon/components/ui/card.tsx:
```diff
@@ -9,7 +9,7 @@ const Card = React.forwardRef<
     ref={ref}
     className={cn(
-      'rounded-lg border bg-card text-card-foreground shadow-sm',
+      'h-full w-full border bg-card text-card-foreground shadow-sm',
```
dont edit the atom, if you need to extend this card into a unique new card, make a molecule that has the extensions that you need.

Balance8 commented on apps/chameleon/components/ui/carousel.tsx:
```diff
@@ -32,6 +33,18 @@ type CarouselContextProps = {
 
 const CarouselContext = React.createContext<CarouselContextProps | null>(null)
 
+const carouselVariant = cva({
```
this is not how we should handle extending the carousel.  You need to take the carousel atoms and make it into your own molecule that has the proper extensions.  See all the other files outside of the UI file.

Balance8 commented on apps/chameleon/components/ui/carousel.tsx:
```diff
@@ -157,12 +170,12 @@ const CarouselContent = React.forwardRef<
 
   return (
-    <div ref={carouselRef} className="overflow-hidden">
+    <div ref={carouselRef} className="overflow-hidden h-full w-full">
```
dont edit the atom

Balance8 commented on apps/chameleon/components/user-auth-form.tsx:
```diff
@@ -30,6 +30,9 @@ export function UserAuthForm({ className, ...props }: UserAuthFormProps) {
   const [isLoading, setIsLoading] = React.useState<boolean>(false)
   const [isGitHubLoading, setIsGitHubLoading] = React.useState<boolean>(false)
   const searchParams = useSearchParams()
+  const variantType: string = props.styleVariant
```
Bringing in props like this breaks the headless model.  If you want to extend the current you need to do so with consideration for the atomic model.

Balance8 commented on apps/chameleon/components/user-auth-form.tsx:
```diff
@@ -80,27 +83,33 @@ export function UserAuthForm({ className, ...props }: UserAuthFormProps) {
       <div className="relative">
         <div className="absolute inset-0 flex items-center">
-          <span className="w-full border-t" />
+          {/* <span className="w-full border-t" /> */}
```
no comments in final product

Balance8 commented on apps/chameleon/components/user-auth-form.tsx:
```diff
@@ -80,27 +83,33 @@ export function UserAuthForm({ className, ...props }: UserAuthFormProps) {
-            Or continue with
-          </span>
+        <div className="relative flex justify-center text-xs font-bold uppercase items-center">
+          <span className="w-[30%] border-t"></span>
```
dont use arb values if they can be avoided I am pretty sure there is a tailwind class for this

Balance8 commented on apps/chameleon/app/(auth)/register/page.tsx:
```diff
@@ -1,44 +1,138 @@
 import { cn } from '@/lib/utils'
 import { buttonVariants } from '@/components/ui/button'
-import { Icons } from '@/components/icons'
+import CarouselRegister from '@/components/signuppage/carouselregister'
```
these imports never made it into the final git commit, you'll need to see that they get added

Balance8 commented on apps/chameleon/app/global.css:
this file needs to be removed. We already have a globals.css file located at apps/chameleon/styles/globals.css

DinDin985 commented:
Added login page and converted carousel to responsive design.
![LoginPage](https://github.com/Ionia-Devs/chameleon/assets/100187027/d766c673-9862-444b-a8c5-067c2372e24f)

Balance8 commented on apps/chameleon/components/ui/card.tsx:
```diff
@@ -1,6 +1,12 @@
+
+
+
+import { cn } from '@/lib/utils';
```
remove unused code

Balance8 commented on apps/chameleon/components/ui/carousel.tsx:
```diff
@@ -1,6 +1,7 @@
 'use client'
 
 import * as React from 'react'
+import { cva } from 'class-variance-authority'
```
remove unused code

Balance8 commented on apps/chameleon/components/ui/input.tsx:
```diff
@@ -10,10 +10,7 @@ const Input = React.forwardRef<HTMLInputElement, InputProps>(
-          'flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50',
-          className
-        )}
+        className={cn('flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50', className)}
```
revert this change

Balance8 commented on apps/chameleon/tailwind.config.js:
```diff
@@ -21,6 +21,8 @@ module.exports = {
     },
     extend: {
       colors: {
+        'dim-black': '#191919',
```
Let's remove this value in favor of a tailwind default black; find one on the colors page that is close, and remove this value from the code base. https://tailwindcss.com/docs/customizing-colors

Balance8 commented on apps/chameleon/tailwind.config.js:
```diff
@@ -21,6 +21,8 @@ module.exports = {
     extend: {
       colors: {
+        'dim-black': '#191919',
+        'neutral-200': '#e5e5e5',
```
this needs to be removed

Balance8 reviewed (changes requested):
@DinDin985  Fix these last few comments and remove the extra globals.css and fix the build errors on the CI/CD pipeline action and then we can merge 👍

DinDin985 commented:
Uploaded the images. But still failed to deploy open to any feedback.

Balance8 reviewed (approved):
Very well done! Huge milestone 🥂

DinDin985 commented:
Mark this day that I got my first major PR. DOPA DOWN!
